server/MVC/models.js

```diff
@@ -0,0 +1,88 @@
+const client = require('../db/connection.js');
+
+module.exports = {
+  // for product
+  getProduct: async (id) => {
+    const product = {
+      name: 'get-product',
+      text: `SELECT json_build_object(
+        'id', $1::INT,
+        'name', name,
+        'slogan', slogan,
+        'description', description,
+        'category', category,
+        'default_price', default_price,
+        'features', (SELECT json_agg
+          (json_build_object(
+            'feature', feature,
+            'value', value
+          ))
+          FROM features
+          WHERE product_id = $1::INT
+        )
+      )
+      AS data
+      FROM products
+      WHERE id = $1`,
+      values: [id]
+    }
+
+    return await client
+      .query(product)
+      .then(({ rows }) => rows[0].data);
+  },
```

Comment:
not in a stage to review this part yet but I will try to come back to it asap :)